dist/misc/uninstall-sgx-powhsm

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# The name of the service unit that will be uninstalled
+SERVICE_NAME=powhsmsgx.service
+# The internal network name for the powHSM service
+NETWORK=powhsmsgx_net
+# The user name created during the installation
+USER=powhsm
+
+function error() {
+    if [ $? -ne 0 ]; then
+        echo "Error: $1"
+        exit 1
+    fi
+}
+
+function promptUserConfirmation() {
+    echo "Are you sure you want to uninstall the powHSM SGX service from this machine?"
+    read -p "Proceed? [y/N]: " confirm
+    if [ "$confirm" != "Y" ] && [ "$confirm" != "y" ]; then
+        echo "Uninstall cancelled."
+        exit 1
+    fi
+}
+
+function backupInstallDirectory() {
+    if [ ! -f "/etc/systemd/system/$SERVICE_NAME" ]; then
+        echo "Service unit file not found, skipping backup."
+        return
+    fi
+
+    WORKING_DIR=$(grep "^WorkingDirectory=" "/etc/systemd/system/$SERVICE_NAME" | cut -d'=' -f2)
+    if [ -z "$WORKING_DIR" ]; then
+        echo "Could not determine working directory from service unit. Aborting."
+        exit 1
+    fi
+
+    if [ ! -d "$WORKING_DIR" ]; then
+        echo "Working directory $WORKING_DIR does not exist, skipping backup."
+        return
+    fi
+
+    if [ -f "backup.tgz" ]; then
+        echo "Error: backup.tgz already exists. Please remove it first."
+        exit 1
+    fi
+
+    echo "Creating backup of $WORKING_DIR..."
+    tar czf backup.tgz -C "$(dirname "$WORKING_DIR")" "$(basename "$WORKING_DIR")"
```

Comment:
Why the `basename` portion? Doesn't that remove any absolute pathing that we'd probably want to keep?

Reply:
And also why the `dirname` portion? Doesn't simply using `tar czf backup.tgz -C $WORKING_DIR $WORKING_DIR` work as expected? Assuming $WORKING_DIR is an absolute path (otherwise we could use `WORKING_DIR=$(realpath $WORKING_DIR)` before actually attempting to create the archive?).

Reply:
I didn't think keeping the entire absolute path was the desired output. I will change it as you suggested, thanks

Reply:
It is not the desired output. The idea is for the archive to be from the perspective of within the installation directory. Meaning we don't want to keep the installation directory name, which can change depending on the installation and is uninteresting. I understand the current approach keeps it, right?

Reply:
> Meaning we don't want to keep the installation directory name, which can change depending on the installation and is uninteresting.

In this case, I believe what we want is `tar czf backup.tgz -C $WORKING_DIR .`, isn't it? In this case we package everything inside `WORKING_DIR` without any parent directories (including the directory name)

Reply:
Yes, thank you.